Source/WebCore/platform/gstreamer/GStreamerElementHarness.cpp

```diff
@@ -185,11 +185,11 @@ void GStreamerElementHarness::start(GRefPtr<GstCaps>&& inputCaps)
     auto streamId = makeString(GST_OBJECT_NAME(m_element.get()), '-', uniqueStreamId.exchangeAdd(1));
     pushEvent(adoptGRef(gst_event_new_stream_start(streamId.ascii().data())));
 
-    pushStickyEvents(WTFMove(inputCaps));
+    pushStickyEvents(WTFMove(inputCaps), WTFMove(segment));
     m_playing.store(true);
 }
 
-void GStreamerElementHarness::pushStickyEvents(GRefPtr<GstCaps>&& inputCaps)
+void GStreamerElementHarness::pushStickyEvents(GRefPtr<GstCaps>&& inputCaps, std::optional<const GstSegment*> segment)
```

Comment:
You're moving the optional into the other method and neither start nor pushStickyEvents take a &&. It is going to work but semantically it has little sense. You might want to reconsider this in one way or another